components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/IdentityProviderManagementServiceTest.java

```diff
@@ -0,0 +1,1077 @@
+        idp1.setIdpProperties(new IdentityProviderProperty[]{identityProviderProperty});
+
+        ClaimConfig claimConfig = new ClaimConfig();
+        claimConfig.setLocalClaimDialect(false);
+        claimConfig.setRoleClaimURI("Country");
+        claimConfig.setUserClaimURI("Country");
+        ClaimMapping cm = ClaimMapping.build("http://wso2.org/claims/country", "Country", "", true);
+        Claim remoteClaim = new Claim();
+        remoteClaim.setClaimId(0);
+        remoteClaim.setClaimUri("Country");
+        claimConfig.setClaimMappings(new ClaimMapping[]{cm});
+        claimConfig.setIdpClaims(new Claim[]{remoteClaim});
+        idp1.setClaimConfig(claimConfig);
+
+        //Initialize Test Identity Provider 2
+        IdentityProvider idp2 = new IdentityProvider();
+        idp2.setIdentityProviderName("testIdP2");
+        idp2.setHomeRealmId("2");
+
+        ClaimConfig claimConfig2 = new ClaimConfig();
+        claimConfig2.setLocalClaimDialect(true);
+        claimConfig2.setRoleClaimURI("http://wso2.org/claims/role");
+        claimConfig2.setUserClaimURI("http://wso2.org/claims/fullname");
+        ClaimMapping cm2 = new ClaimMapping();
+        Claim localClaim2 = new Claim();
+        localClaim2.setClaimId(0);
+        localClaim2.setClaimUri("http://wso2.org/claims/fullname");
+        cm2.setLocalClaim(localClaim2);
+        claimConfig2.setClaimMappings(new ClaimMapping[]{cm2});
+        idp2.setClaimConfig(claimConfig2);
+
+        //Initialize Test Identity Provider 3
+        IdentityProvider idp3 = new IdentityProvider();
+        idp3.setIdentityProviderName("testIdP3");
+        idp3.setHomeRealmId("3");
+
+        /* IDP with PermissionsAndRoleConfig,FederatedAuthenticatorConfigs, ProvisioningConnectorConfigs,ClaimConfigs */
+        identityProviderManagementService.addIdP(idp1);
+
+        /* IDP with Local Cliam Dialect ClaimConfigs */
+        identityProviderManagementService.addIdP(idp2);
+
+        /* IDP with Only name */
+        identityProviderManagementService.addIdP(idp3);
+    }
+
+    private void addResidentIdp(IdentityProviderManagementService identityProviderManagementService) throws
+            IdentityProviderManagementException {
+
+        IdentityProvider residentIdp = new IdentityProvider();
+        residentIdp.setIdentityProviderName("LOCAL");
+        IdentityProviderProperty idpProperty1 = new IdentityProviderProperty();
+        idpProperty1.setName(IdentityApplicationConstants.SESSION_IDLE_TIME_OUT);
+        idpProperty1.setValue("20");
+        residentIdp.setIdpProperties(new IdentityProviderProperty[]{idpProperty1});
+
+        identityProviderManagementService.addIdP(residentIdp);
+    }
+
+    private void addSharedIdp() throws SQLException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+
+            String sql = "INSERT INTO IDP (TENANT_ID, NAME, UUID, IS_PRIMARY, IS_FEDERATION_HUB, " +
```

Comment:
Can't we reuse the same queries used in the DAO layer?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/3493/commits/6b030460eec87be9f79276256c9a3617fe357398